db.go

```diff
@@ -379,6 +379,27 @@ func (db *DB) compact() (changes bool, err error) {
 		}
 		changes = true
 
+		// close blocks in plan so that we can remove files.
+		var blocks []*Block
+		db.mtx.Lock()
+		oldBlocks := db.blocks
+		for _, b := range oldBlocks {
+			keep := true
+			for _, pd := range plan {
+				if pd == b.Dir() {
+					keep = false
+					break
+				}
+			}
+			if keep {
+				blocks = append(blocks, b)
+			} else {
+				b.Close()
+			}
+		}
+		db.blocks = blocks
+		db.mtx.Unlock()
```

Comment:
This will cause data for the given time range to become unavailable until the reload, which is called later gets completed.

Also, introspecting `db.blocks` in this code path kinda throws a few things together that were separated before.

But I suppose windows not allowing us to delete open files literally forces us to have this small blip where data is not accessible. Anyone sees a way around this?